src/transaction/core/useGetTransactionStatus.tsx

```diff
@@ -0,0 +1,45 @@
+import { ReactNode, useMemo } from 'react';
+import { useTransactionContext } from '../components/TransactionProvider';
+import { cn, color, text } from '../../styles/theme';
+
+export function useGetTransactionStatus() {
+  const { errorMessage, isLoading, transactionId } = useTransactionContext();
+
+  return useMemo(() => {
+    let actionElement: ReactNode = null;
+    let label = 'Complete the required fields to continue';
+    let labelClassName: string = color.foregroundMuted;
+
+    if (isLoading) {
+      label = 'Transaction in progress...';
+      actionElement = (
+        <a>
+          <span className={cn(text.label1, color.primary)}>
+            View on explorer
+          </span>
+        </a>
+      );
+    }
+    if (transactionId) {
+      label = 'Successful!';
+      actionElement = (
+        <a href={`https://basescan.org/tx/${transactionId}`} target="_blank">
```

Comment:
to remember `href` will change based on the Chain.

We can create an utility called `getChainExplorer`, and have it at top level similar on how we have `isBase`, and that can be an utility to use across all modules.